### PR TITLE
Fixing option encoding

### PR DESCRIPTION
**Pull request by JorisGoosen**
Fixes https://github.com/jasp-stats/INTERNAL-jasp/issues/1163 (ANOVA extra encoding broken)
Fixes https://github.com/jasp-stats/INTERNAL-jasp/issues/1162 (JAGS gets confused about columnnames)

### CODE REVIEW

**Comment by boutinb** [comment]
I have tested the file in https://github.com/jasp-stats/INTERNAL-jasp/issues/1163 (RepMeasuresANOVA_SPACEY). 
It seems to work in 0.14.
In development I get this error:
Screenshot 2020-11-13 at 14 26 51

With your branch I get:
Screenshot 2020-11-13 at 14 27 04

I'm not sure if this is better!

Furthermore, the encoding is set in BoundQMLListViewTerms which is used by many kinds of controls. For example the ComponentsList used in all MixedModels analyses. And if I test a MixedModels analysis  I get the same error:
`Trying to decode columnName but 'Kangaroo Testicle' is not an encoded columnName!'`

Maybe it's better to wait for the Control refactor project is finished, and then to handle this problem.

**Comment by JorisGoosen** [comment]
Did you do a proper clean and all that?

**Comment by JorisGoosen** [comment]
And please lets merge this first before the whole refactor...
JAGS is fixed for sure btw.

I can check later whether I fixed it properly or not.

But all those `Trying to decode columnName but 'Kangaroo Testicle' is not an encoded columnName!'` errors are what I was trying to fix...

I also don't get what you mean by "tested in `development`", you should test my branch right?

File: Engine/engine.cpp [hunk 1]
```diff
@@ -501,31 +505,48 @@ void Engine::_encodeColumnNamesinOptions(Json::Value & options, Json::Value & me
...
 				_encodeColumnNamesinOptions(options[i], meta[i]);
-		else if(meta.type() == Json::objectValue && meta.get("containsColumn", false).asBool())
+		
+		else if(isRCode)
+		{
+			for(size_t i=0; i<options.size(); i++)
+				if(options[i].isString())
+					options[i] = ColumnEncoder::columnEncoder()->encodeRScript(options[i].asString());
+		}
+		else if(encodePlease)
```
Comment (boutinb) [inline Engine/engine.cpp]: This extra `else if(encodePlease)` is quite redundant.

**Review by boutinb — CHANGES_REQUESTED** [review]
Just a small remark.
It seems to work indeed.

File: Engine/engine.cpp (hunk 1 above)
Comment (JorisGoosen) [reply Engine/engine.cpp]: Indeed indeed, ive fixed it now. Not sure how I missed that before but ah well :p

**Pull request merged**